bin/cron/sync_dropbox

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env ruby
+
+# This syncs content changes from the shared dropbox folder to our github repo
+# on the staging machine. This is required for contents changes from dropbox to
+# be deployed.
+
+require_relative '../../lib/cdo/only_one'
+exit unless only_one_running?(__FILE__)
+require_relative '../../dashboard/config/environment'
+exit unless rack_env?(:staging) && CDO.dashboard_hostname == 'staging-studio.code.org'
+
+FOLDERS = {
+  "advocacy.code.org" => "sites.v3/advocacy.code.org",
+  "applab-docs-1" => "sites.v3/code.org/public/applab/docs1",
+  "applab-docs" => "sites.v3/code.org/public/applab/docs",
+  "code.org" => "sites.v3/code.org",
+  "csedweek.org" => "sites.v3/csedweek.org",
+  "curriculum-6-8" => "sites/virtual/curriculum-6-8",
+  "curriculum-algebra" => "sites/virtual/curriculum-algebra",
+  "curriculum-course1" => "sites/virtual/curriculum-course1",
+  "curriculum-course2" => "sites/virtual/curriculum-course2",
+  "curriculum-course3" => "sites/virtual/curriculum-course3",
+  "curriculum-course4" => "sites/virtual/curriculum-course4",
+  "curriculum-csp" => "sites/virtual/curriculum-csp",
+  "curriculum-docs" => "sites/virtual/curriculum-docs",
+  "curriculum-misc" => "sites/virtual/curriculum-misc",
+  "curriculum-science" => "sites/virtual/curriculum-science",
+  "curriculum-unplugged" => "sites/virtual/curriculum-unplugged",
+  "emails" => "emails",
+  "hourofcode.com" => "sites.v3/hourofcode.com"
+}.freeze
+
+FOLDERS.each do |key, value|
+  # 'unison' will sync the two folders. It will return true on success and false on failure.
+  # For full details, see unison's man page. Docs are also here: https://www.cis.upenn.edu/~bcpierceg/unison/
+  command = "unison /home/ubuntu/Dropbox/Shared/#{key} /home/ubuntu/staging/pegasus/#{value} -silent -ignoreinodenumbers -ignore \"Name .dropbox\" -auto -perms 0 -dontchmod"
+  if key == "code.org"
+    command << " -ignore \"Path */public/applab/docs\" -ignore \"Path */public/applab/docs1\""
```

Comment:
Can you explain this in a comment? (I know why but it'll probably look pretty mysterious to others.)

Reply:
👍